test/cloudant.test.js

```diff
@@ -112,6 +112,37 @@ describe('cloudant connector', function() {
         };
       });
   });
+
+  describe('updateOrCreate', function() {
+    afterEach(function(done) {
+      Product.destroyById(1, function(err) {
+        if (err) return done(err);
+        done();
+      });
+    });
+
+    it('can call upsert with an id and have it create a document without error',
```

Comment:
Probably the following is what you are achieving here:

```
can call upsert with an a primary key and create/update a document without error
```

Also it is always a good idea if you try primary keys which are not `id` as well.